src/Settings/Builders/RunTest.hs

```diff
@@ -81,53 +72,85 @@ runTestBuilderArgs = builder RunTest ? do
+        junitArg     = case testJUnit args of
+                           Just filepath -> Just $ "--junit " ++ quote filepath
+                           Nothing -> Nothing
+        configArgs   = concat [["-e", configArg] | configArg <- testConfigs args]
         verbosityArg = case testVerbosity args of
-                         Nothing -> Nothing
-                         Just verbosity -> Just $ "--verbose=" ++ verbosity
-        wayArgs    = map ("--way=" ++) (testWays args)
-    pure $  testOnlyArg
-         ++ speedArg
+                           Nothing -> Just $ "--verbose=" ++ show (fromEnum globalVerbosity)
+                           Just verbosity -> Just $ "--verbose=" ++ verbosity
+        wayArgs      = map ("--way=" ++) (testWays args) 
+        compilerArg  = ["--config", "compiler=" ++ show (compiler)]
+        ghcPkgArg    = ["--config", "ghc_pkg=" ++ show (bindir -/- "ghc-pkg")]
+        haddockArg   = ["--config", "haddock=" ++ show (bindir -/- "haddock")]
+        hp2psArg     = ["--config", "hp2ps=" ++ show (bindir -/- "hp2ps")]
+        hpcArg       = ["--config", "hpc=" ++ show (bindir -/- "hpc")]   
+    pure $  testOnlyArg ++ speedArg 
          ++ catMaybes [ onlyPerfArg, skipPerfArg, summaryArg
-                      , junitArg, verbosityArg  ]
-         ++ configArgs
-         ++ wayArgs
+                      , junitArg, verbosityArg  ] 
+         ++ configArgs ++ wayArgs ++  compilerArg ++ ghcPkgArg
+         ++ haddockArg ++ hp2psArg ++ hpcArg
+
+-- | Directory to look for Binaries
+-- | We assume that required programs are present in the same binary directory 
+-- | in which ghc is stored and that they have their conventional name.
+-- | QUESTION : packages can be named different from their conventional names.
+-- | For example, ghc-pkg can be named as ghc-pkg-version. In such cases, it will
+-- | be impossible to search the binary. Only possible way will be to take user 
+-- | inputs for these directory also. boilerplate soes not account for this 
+-- | problem, but simply returns an error. How should we handle such cases?
+setBinaryDirectory :: String -> Action FilePath
+setBinaryDirectory "stage0" = setting InstallBinDir
+setBinaryDirectory "stage1" = liftM2 (-/-) topDirectory (stageBinPath Stage0) 
+setBinaryDirectory "stage2" = liftM2 (-/-) topDirectory (stageBinPath Stage1) 
+setBinaryDirectory compiler = pure $ parentPath compiler
+
+-- | Set Test Compiler
+setCompiler :: String -> Action FilePath
+setCompiler "stage0" = setting SystemGhc
+setCompiler "stage1" = liftM2 (-/-) topDirectory (fullpath Stage0 ghc)
+setCompiler "stage2" = liftM2 (-/-) topDirectory (fullpath Stage1 ghc)
+setCompiler compiler = pure compiler 
 
 -- | Set speed for test
 setTestSpeed :: TestSpeed -> String
 setTestSpeed Fast    = "2"
 setTestSpeed Average = "1"
 setTestSpeed Slow    = "0"
 
+-- | Returns parent path of test compiler 
+-- | TODO : Can this be implemented in a better way
```

Comment:
Could you be more specific here? Describe why the current approach is not good enough, so that it is clear to the person comes across this `TODO` what the end end goal is.

Reply:
Actually, I just find it a bit hacky. I cannot think of any reason why this will fail. Just the approach is not satisfying in my opinion.